e2e/pages/citizen/createCase/C100/casePartyDetails/applicantAddressLookupPage.ts

```diff
@@ -0,0 +1,111 @@
+import AccessibilityTestHelper from "../../../../../common/accessibilityTestHelper";
+import { Page } from "@playwright/test";
+import { Selectors } from "../../../../../common/selectors";
+import { ApplicantAddressLookupContent } from "../../../../../fixtures/citizen/createCase/C100/casePartyDetails/applicantAddressLookupContent";
+import { Helpers } from "../../../../../common/helpers";
+import { CommonStaticText } from "../../../../../common/commonStaticText";
+
+interface applicantAddressLookupOptions {
+  page: Page;
+  accessibilityTest: boolean;
+  errorMessaging: boolean;
+  addressLookup: boolean;
+}
+
+interface checkPageLoadsOptions {
+  page: Page;
+  accessibilityTest: boolean;
+}
+
+interface fillInFieldsOptions {
+  page: Page;
+  errorMessaging: boolean;
+  addressLookup: boolean;
+}
+
+enum inputIds {
+  postcodeLookup = '#addressPostcode',
+  manualAddress = '#enterAddressManually',
+}
+
+export class ApplicantAddressLookupPage {
+  public static async applicantAddressLookupPage({
+                                                         page,
+                                                         accessibilityTest,
+                                                         errorMessaging, addressLookup,
+                                                       }: applicantAddressLookupOptions): Promise<void> {
+    await this.checkPageLoads({ page, accessibilityTest });
+    if (errorMessaging) {
+      await this.triggerErrorMessages(page);
+    }
+    await this.fillInFields({
+      page,
+      errorMessaging,
+      addressLookup,
+    });
+  }
+
+  private static async checkPageLoads({
+                                        page,
+                                        accessibilityTest,
+                                      }: checkPageLoadsOptions): Promise<void> {
+    await page.waitForSelector(
+      `${Selectors.GovukHeadingL}:text-has("${ApplicantAddressLookupContent.pageTitle}")`
```

Comment:
I think the playwright pseudo-selector is `has-text`, but this may be one I've not come across